src/risclog/sqlalchemy/tests/test_cache.py

```diff
@@ -0,0 +1,253 @@
+from .. import model
+from ..cache import ModelCache
+from ..cache import MultipleObjectsFoundException
+from sqlalchemy import Column
+from sqlalchemy import ForeignKey
+from sqlalchemy import Integer
+from sqlalchemy import String
+import pytest
+import sqlalchemy
+
+
+class TestObject(model.ObjectBase):
+    _engine_name = 'db3'
+
+
+Object = model.declarative_base(TestObject)
+
+
+class PlainModel(Object):
+    id = Column(String(10), primary_key=True)
+    titel = Column(String(100))
+
+
+class LinkedModel(Object):
+    id = Column(String(10), primary_key=True)
+    sequence_model_id = Column(
+        Integer,
+        ForeignKey('sequencemodel.id'),
+        primary_key=True,
+    )
+    sequence_model = sqlalchemy.orm.relation('SequenceModel', uselist=False)
+
+
+class SequenceModel(Object):
+    id = Column(Integer, autoincrement=True, primary_key=True)
+    titel = Column(String)
+
+
+@pytest.fixture(scope='session')
+def db(database_3):
+    database_3.create_all('db3')
+    return database_3
```

Comment:
```suggestion
    yield database_3
```
We can now always yield in fixtures.